src/app/plugins/quote.plugin.ts

```diff
@@ -0,0 +1,93 @@
+import Constants from '../../common/constants';
+import { Plugin } from '../../common/plugin';
+import { ChannelType, IContainer, IHttpResponse, IMessage } from '../../common/types';
+import { MessageEmbed } from 'discord.js';
+
+export default class QuotePlugin extends Plugin {
+  public commandName: string = 'quote';
+  public name: string = 'Quote Plugin';
+  public description: string = 'Prints a random quote';
+  public usage: string = 'quote <tag (optional)>';
+  public override pluginAlias = [];
+  public permission: ChannelType = ChannelType.Public;
+  public override pluginChannelName: string = Constants.Channels.Public.General;
+
+  private _DEFAULT_URL: string = 'https://api.quotable.io/random';
+  private _VALID_TAGS: string[] = [
+    'business',
+    'education',
+    'faith',
+    'famous-quotes',
+    'friendship',
+    'future',
+    'happiness',
+    'history',
+    'inspirational',
+    'life',
+    'literature',
+    'love',
+    'nature',
+    'politics',
+    'proverb',
+    'religion',
+    'science',
+    'success',
+    'technology',
+    'wisdom',
+  ];
+
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  private _createEmbed(content: string, author: string, tag: string) {
+    const embed: MessageEmbed = new MessageEmbed()
+      .setColor('#0099ff')
+      .setTitle(`${tag}`)
+      .setFooter('Pulled from api.quotable.io')
+      .setTimestamp(new Date())
+      .setDescription(
+        `${content}
+
+        - ${author}`
+      );
+
+    if (tag === 'Famous-quotes') embed.setTitle('Famous-Quotes');
+
+    return embed;
+  }
+
+  public async execute(message: IMessage, args: string[]) {
+    let url: string = this._DEFAULT_URL;
+
+    if (args.length > 1) {
+      await message.reply('Too many arguments, please try again.');
+      return;
+    }
+
+    if (args.length > 0 && args.length < 2) {
+      if (!this._VALID_TAGS.includes(args[0].toLowerCase())) {
+        await message.reply('Invalid tag, please try again.');
+        return;
+      }
+      url = `${url}?tags=${args[0].toLowerCase()}`;
+    }

```

Comment:
Use plugin.validate for validation
[Override this](https://github.com/cs-discord-at-ucf/lion/blob/5ac3e93821667309d981baea9726e837540110a0/src/common/plugin.ts#L35-L41)